### ISSUE TITLE
Option to force the Player to wake up when Fatigue hits 0

### ISSUE THREAD

**Issue by Geekerandy** [Su5eD/Somnia-Awoken#71]
### Is your feature request related to a problem? Please describe.

Simply put, I wanted to balance the night-skipping benefits of sleep in a modpack where the average night is longer than usual (or varies, such as B3M mod)

However, there is no setting to scale how long a player will sleep in regards to their fatigue level. A player will still sleep the length of one night, whether at 20 fatigue, or 100 fatigue. 

Surprisingly enough, despite the noted "undesired behavior" from 13uRn on the CurseForge page about incompatibilities with B3M mod, it was going in the right direction for what I wanted.

So my trouble is that players sleep too often, and get off easy, rather than face the threats that appear in the night.
And then at the extreme end, I can of course have the fatigue effects come in for not sleeping.

### Describe the solution you'd like

Basically, I want "sleep" as a resource, where fatigue is really a mechanic of "how much time you can skip safely"

So really, all I would need is a "force player to wake up when Fatigue hits 0."

I can adjust the other tuning values to how I see fit for my pack.
